### ISSUE TITLE
Make use of the element_count annotation

### ISSUE THREAD

**Issue by jvoisin** [jvoisin/fortify-headers#11]
- clang: https://reviews.llvm.org/D148381
- gcc: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=108896

**Comment by nekopsykose** [comment]
something i noticed is that for things like e.g. fread

```c
size_t fread(void ptr[restrict .size * .nmemb],
             size_t size, size_t nmemb,
             FILE *restrict stream);
```

this case is seemingly not supported (can't element_count(size*nmemb) ?)

**Comment by nekopsykose** [comment]
actually since it's for fams only it's not useful for that anyway

**Comment by nekopsykose** [comment]
hm, or is it? all the discussion talks about using it for fams, but if it's just a builtin_object_size complement then it should be usable on function args too...

**Comment by jvoisin** [comment]
Yeah, the current status of `element_count` is a bit confusing unfortunately. Let's wait until the dust settles and for the Linux kernel to move forward with it :P

**Comment by kees** [comment]
`counted_by` only currently works for flexible array members. https://opensource.googleblog.com/2024/07/bounds-checking-flexible-array-members.html
Coming next will be applying it to pointer members and for function arguments.
